### ISSUE TITLE
Remove Google Test

### ISSUE THREAD

**Issue by lookeypl** [mkulagowski/MineZPRft#9]
It looks like we shouldn't keep Google Test on our repository. Remove it from repository and provide instructions on how to build it and install on both platforms inside README.md.

**Comment by lookeypl** [comment]
On Ubuntu `sudo apt-get install libgtest-dev` should be enough

**Comment by mkulagowski** [comment]
I am able to install it using this command on Mint Qiana as well. Well done.

**Comment by mkulagowski** [comment]
Problems:
`sudo apt-get install libgtest-dev` does not work - it only downloads sources and headers. It does not download static lib, because some time ago google said "naah, we don't want to distribute static libs" - [here](http://www.ubuntuupdates.org/package/core/precise/universe/base/libgtest-dev) You can check changes in the package.

In that case we can use a Cmake-way preferred by google - [using "External Project" module](http://www.kaizou.org/2014/11/gtest-cmake/).
It downloads gtest as external project, builds it and links as a library.
Here is some other working example of using it:

```
cmake_minimum_required (VERSION 3.1)

project (registerer)

##################################
# Download and install GoogleTest

include(ExternalProject)
ExternalProject_Add(gtest
  URL https://googletest.googlecode.com/files/gtest-1.7.0.zip
  # Comment above line, and uncomment line below to use subversion.
  # SVN_REPOSITORY http://googletest.googlecode.com/svn/trunk/ 
  # Uncomment line below to freeze a revision (here the one for 1.7.0)
  # SVN_REVISION -r700

  PREFIX ${CMAKE_CURRENT_BINARY_DIR}/gtest
  INSTALL_COMMAND ""
)
ExternalProject_Get_Property(gtest source_dir binary_dir)

################
# Define a test
add_executable(registerer_test registerer_test.cc)

######################################
# Configure the test to use GoogleTest
#
# If used often, could be made a macro.

add_dependencies(registerer_test gtest)
include_directories(${source_dir}/include)
target_link_libraries(registerer_test ${binary_dir}/libgtest.a)
target_link_libraries(registerer_test ${binary_dir}/libgtest_main.a)

##################################
# Just make the test runnable with
#   $ make test

enable_testing()
add_test(NAME    registerer_test 
         COMMAND registerer_test)
```

I've also found [another way](http://askubuntu.com/questions/145887/why-no-library-files-installed-for-google-test). Here we just apt-get the package, then build and move *.a files.
Then we can use `target_link_libraries` just the same way as above.

**Comment by lookeypl** [comment]
In that case we could as well leave the sources as they are. OR, use git submodule to attach gtest to our repository.

I have no f*cking idea now... Let me think about this for a second, later in the evening we will decide what to do with it.

**Comment by lookeypl** [comment]
OK, after some thinking, I am changing the priority on this one to a low one.

At worst, we will lose some points. But this is a very minor problem for us right now IMO we do not need to fix this right away - it is a separate piece of code, which can be fixed anytime (unlike #8 ).

Feel free to leave this task. We have more important things to do.

**Comment by lookeypl** [comment]
The project was given to evaluate, gtest still stays within our repository. No need to change that. Closing
